Library/Homebrew/formula_auditor.rb

```diff
@@ -171,6 +171,12 @@ def audit_formula_name
       problem "Formula name conflicts with existing core formula."
     end
 
+    def audit_formula_revision
```

Comment:
What do you think about rolling this into `audit_revision_and_version_scheme`? It could be right at the beginning before anything else.

Reply:
I looked there when I was finally starting to get the hang of this Ruby (not in my top n languages).

The code at least was similar in the original place I misplaced it, but the code felt so different from `audit_revision_and_version_scheme` that I just put it next to it.

I'm not opposed to putting it inside. Is there anything special I need to do?

This is what I ended up with (the jitter is from `brew style --fix`):

```diff
diff --git a/Library/Homebrew/formula_auditor.rb b/Library/Homebrew/formula_auditor.rb
index 7edfaf0a1..8041908da 100644
--- a/Library/Homebrew/formula_auditor.rb
+++ b/Library/Homebrew/formula_auditor.rb
@@ -171,12 +171,6 @@ module Homebrew
       problem "Formula name conflicts with existing core formula."
     end
 
-    def audit_formula_revision
-      return unless @new_formula
-
-      new_formula_problem("New formulae should not define a revision.") unless formula.revision.zero?
-    end
-
     PERMITTED_LICENSE_MISMATCHES = {
       "AGPL-3.0" => ["AGPL-3.0-only", "AGPL-3.0-or-later"],
       "GPL-2.0"  => ["GPL-2.0-only",  "GPL-2.0-or-later"],
@@ -680,6 +674,8 @@ module Homebrew
     end
 
     def audit_revision_and_version_scheme
+      new_formula_problem("New formulae should not define a revision.") if @new_formula && !formula.revision.zero?
+
       return unless @git
       return unless formula.tap # skip formula not from core or any taps
       return unless formula.tap.git? # git log is required
diff --git a/Library/Homebrew/test/dev-cmd/audit_spec.rb b/Library/Homebrew/test/dev-cmd/audit_spec.rb
index 41ee0faa8..27560df7f 100644
--- a/Library/Homebrew/test/dev-cmd/audit_spec.rb
+++ b/Library/Homebrew/test/dev-cmd/audit_spec.rb
@@ -853,36 +853,13 @@ module Homebrew
       end
     end
 
-    describe "#audit_formula_revision" do
-      it "doesn't allow new formulae to have a revision" do
-        fa = formula_auditor "foo", <<~RUBY, new_formula: true
-          class Foo < Formula
-            url "https://brew.sh/foo-1.0.tgz"
-            revision 1
-          end
-        RUBY
-
-        fa.audit_formula_revision
-
-        expect(fa.new_formula_problems).to include(
-          a_hash_including(message: a_string_matching(/should not define a revision/)),
-        )
-      end
-    end
-
     describe "#audit_revision_and_version_scheme" do
-      subject {
-        fa = described_class.new(Formulary.factory(formula_path), git: true)
-        fa.audit_revision_and_version_scheme
-        fa.problems.first&.fetch(:message)
-      }
-
-      let(:origin_tap_path) { Tap::TAP_DIRECTORY/"homebrew/homebrew-foo" }
-      let(:foo_version) { Count.increment }
-      let(:formula_subpath) { "Formula/foo#{foo_version}.rb" }
-      let(:origin_formula_path) { origin_tap_path/formula_subpath }
-      let(:tap_path) { Tap::TAP_DIRECTORY/"homebrew/homebrew-bar" }
       let(:formula_path) { tap_path/formula_subpath }
+      let(:tap_path) { Tap::TAP_DIRECTORY/"homebrew/homebrew-bar" }
+      let(:origin_formula_path) { origin_tap_path/formula_subpath }
+      let(:formula_subpath) { "Formula/foo#{foo_version}.rb" }
+      let(:foo_version) { Count.increment }
+      let(:origin_tap_path) { Tap::TAP_DIRECTORY/"homebrew/homebrew-foo" }
 
       before do
         origin_formula_path.dirname.mkpath
@@ -908,6 +885,29 @@ module Homebrew
         end
       end
 
+      describe "new formulae should not have a revision" do
+        it "doesn't allow new formulae to have a revision" do
+          fa = formula_auditor "foo", <<~RUBY, new_formula: true
+            class Foo < Formula
+              url "https://brew.sh/foo-1.0.tgz"
+              revision 1
+            end
+          RUBY
+
+          fa.audit_revision_and_version_scheme
+
+          expect(fa.new_formula_problems).to include(
+            a_hash_including(message: a_string_matching(/should not define a revision/)),
+          )
+        end
+      end
+
+      subject {
+        fa = described_class.new(Formulary.factory(formula_path), git: true)
+        fa.audit_revision_and_version_scheme
+        fa.problems.first&.fetch(:message)
+      }
+
       def formula_gsub(before, after = "")
         text = formula_path.read
         text.gsub! before, after
```